### ISSUE TITLE
Editor panics when opening empty files.

### ISSUE THREAD

**Issue by mmatongo** [curlpipe/ox#28]
**Describe the bug**
Opening empty files causes `main` to panic printing out the error;
 ```
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', src/editor.rs:318:17
```
**To Reproduce**

1. `touch foo.c` (or any extension)
2. `xo foo.c`
3. Press any key and the error should pop up.

**Expected behavior**
I expect the editor to go into insert mode but instead this happens.

**Screenshots**
[A screen cast](https://asciinema.org/a/KaMnwckr6ArE0seak7QBavYEM)

**Desktop (please complete the following information):**
 - OS: KISS Linux (KISS is a musl based OS.)
 - Kernal: 5.4.65

**Comment by curlpipe** [comment]
Ah yes, I spotted this earlier today, i'll be sure to fix it :smile:

**Comment by mmatongo** [comment]
> Ah yes, I spotted this earlier today, i'll be sure to fix it

Awesome, I'm loving the editor so far and It's replaced `neovim` as my note taking app in the meantime.
It's so straight forward.

**Comment by curlpipe** [comment]
Hi! Thank you so much for reporting this issue, I've since fixed it in commit 9541105
:+1:

**Comment by mmatongo** [comment]
Thank you 🙏
